ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -126,21 +126,30 @@ public int execute() {
       Path dumpRoot = new Path(conf.getVar(HiveConf.ConfVars.REPLDIR),
               Base64.getEncoder().encodeToString(work.dbNameOrPattern.toLowerCase()
                       .getBytes(StandardCharsets.UTF_8.name())));
-      Path currentDumpPath = new Path(dumpRoot, getNextDumpDir());
-      Path hiveDumpRoot = new Path(currentDumpPath, ReplUtils.REPL_HIVE_BASE_DIR);
-      DumpMetaData dmd = new DumpMetaData(hiveDumpRoot, conf);
-      // Initialize ReplChangeManager instance since we will require it to encode file URI.
-      ReplChangeManager.getInstance(conf);
-      Path cmRoot = new Path(conf.getVar(HiveConf.ConfVars.REPLCMDIR));
-      Long lastReplId;
-      if (!dumpRoot.getFileSystem(conf).exists(dumpRoot)
-              || dumpRoot.getFileSystem(conf).listStatus(dumpRoot).length == 0) {
-        lastReplId = bootStrapDump(hiveDumpRoot, dmd, cmRoot, hiveDb);
+      Path previousDumpMetaPath = getPreviousDumpMetadataPath(dumpRoot);
```

Comment:
Does this mean that. user is not supposed to Alter the dump directory in the middle of replication? Any way to restrict that operation?

Reply:
yes only hive user should have access to this staging directory as we are completely relying on FS based ack.
But in case of irrecoverable users, users should delete the _load_execption file for dump/load to proceed. So they will need some admin access. But in general they shouldn't modify anything.